### PR TITLE
BUG: Fix Index.__repr__ when `display.max_seq_items` = 1

### LINKED ISSUE #38415: BUG: MultiIndex.__repr__ is broken when `display.max_seq_items` = 1
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
In[11]: pd.set_option("display.max_seq_items", 2)
In[12]: pd.MultiIndex.from_tuples(tuples)
Out[12]: 
MultiIndex([(1,  'red'),
            ...
            (2, 'blue')],
           length=40)
In[13]: tuples = [(1, 'red'), (1, 'blue'), (2, 'red'), (2, 'blue')]
In[14]: pd.MultiIndex.from_tuples(tuples)
Out[14]: 
MultiIndex([(1,  'red'),
            ...
            (2, 'blue')],
           length=4)
In[15]: pd.set_option("display.max_seq_items", 1)
In[16]: pd.MultiIndex.from_tuples(tuples)
Out[16]: 
MultiIndex([...
            (1,  'red'),
            (1, 'blue'),
            (2,  'red'),
            (2, 'blue')],
           length=4)
```

#### Problem description

The output of `MultiIndex.__repr__` is dependent on `display.max_seq_items`, the output is correct for different values of `max_seq_items` except when it is set to 1. When it is set to 1, the entire records in MultiIndex are being printed and also `"..."` are being prepended to the data output.

#### Expected Output
```python
In[12]: pd.MultiIndex.from_tuples(tuples)
Out[12]: 
MultiIndex([(1,  'red'),
            ...],
           length=40)
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : b5958ee1999e9aead1938c0bba2b674378807b3d
python           : 3.7.9.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-53-generic
Version          : #59-Ubuntu SMP Wed Oct 21 09:38:44 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8
pandas           : 1.1.5
numpy            : 1.19.4
pytz             : 2020.4
dateutil         : 2.8.1
pip              : 20.3.1
setuptools       : 49.6.0.post20201009
Cython           : 0.29.21
pytest           : 6.1.2
hypothesis       : 5.43.3
sphinx           : 3.3.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.19.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : 0.8.4
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 1.0.1
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : 0.52.0

</details>

### PR DESCRIPTION
**Pull request by skvrahul**
- [x] closes #38415 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
Thanks @skvrahul for the PR. please add a test.

File: doc/source/whatsnew/v1.3.0.rst [hunk 1]
```diff
@@ -75,8 +75,10 @@ Performance improvements
 Bug fixes
 ~~~~~~~~~
 
--
--
+I/O
```
Comment (jreback) [inline doc/source/whatsnew/v1.3.0.rst:78]: there is an I/O section below

File: pandas/io/formats/printing.py [hunk 2]
```diff
@@ -382,7 +382,11 @@ def best_len(values: List[str]) -> int:
         summary = f"[{first}, {last}]{close}"
     else:
 
-        if n > max_seq_items:
+        if max_seq_items == 1:
```
Comment (jreback) [inline pandas/io/formats/printing.py:385]: can you also cover in a test the max_seq_items == 0 case (I think we have this but maybe not). Also pls confirm that we have a test for n == max_seq_items.

**Review by jreback — CHANGES_REQUESTED** [review]
this issue needs to confirm / add any tests that we are missing

File: doc/source/whatsnew/v1.3.0.rst (hunk 1 above)
Comment (skvrahul) [reply doc/source/whatsnew/v1.3.0.rst:78]: must have missed that. Will move it there

File: pandas/io/formats/printing.py (hunk 2 above)
Comment (skvrahul) [reply pandas/io/formats/printing.py:385]: Looks like we don't have a test for `n == max_seq_items`. Will cover these in my tests. 

Just wanted to confirm that this is the relevant location for adding these?
https://github.com/pandas-dev/pandas/blob/e2dec8d03020eb2b31d376a7fcd7348977c6a319/pandas/tests/indexes/multi/test_formats.py#L92-L119

**Comment by skvrahul** [comment]
@jreback @simonjayhawkins 
Could you confirm that this parameter is expected to affect even the list of names of columns in the index
I.e. `repr` evaluates to:
```
MultiIndex([...
            ('qux', 'two')],
           names=['first', ...], length=6)
```
and not
```
MultiIndex([...
            ('qux', 'two')],
           names=['first', 'second'], length=6)
```

**Comment by skvrahul** [comment]
> Could you confirm that this parameter is expected to affect even the list of names of columns in the index
> I.e. `repr` evaluates to:
> 
> ```
> MultiIndex([...
>             ('qux', 'two')],
>            names=['first', ...], length=6)
> ```
> 
> and not
> 
> ```
> MultiIndex([...
>             ('qux', 'two')],
>            names=['first', 'second'], length=6)
> ```

Checked on `master` and can confirm this (parameter affects even the list of column names)
Added tests accordingly.

File: pandas/io/formats/printing.py (hunk 2 above)
Comment (jreback) [reply pandas/io/formats/printing.py:385]: yeah there plus some generic Index ones (look in pandas/tests/indexes/); you may have to grep around a bit

e.g.

```
~/pandas-dev$ grep -r max_seq_items pandas/tests/
pandas/tests/io/formats/test_format.py:        with option_context("display.max_seq_items", 2000):
pandas/tests/io/formats/test_format.py:        with option_context("display.max_seq_items", 5):
pandas/tests/groupby/test_groupby.py:    "max_seq_items, expected",
pandas/tests/groupby/test_groupby.py:def test_groups_repr_truncates(max_seq_items, expected):
pandas/tests/groupby/test_groupby.py:    with pd.option_context("display.max_seq_items", max_seq_items):
pandas/tests/indexes/multi/test_formats.py:    with pd.option_context("display.max_seq_items", None):
pandas/tests/indexes/multi/test_formats.py:        with pd.option_context("display.max_seq_items", 5):
pandas/tests/indexes/common.py:        with pd.option_context("display.max_seq_items", None):
pandas/tests/indexes/base_class/test_formats.py:        with cf.option_context("display.max_seq_items", 10):
```

i think `pandas/tests/indexes/base_class/test_formats.py:        with cf.option_context("display.max_seq_items", 10):`
has a number of tests as well

File: pandas/tests/indexes/multi/test_formats.py [hunk 3]
```diff
@@ -118,6 +118,28 @@ def test_repr(self, idx):
            names=['first', 'second'], length=6)"""
             assert result == expected
 
+        # display.max_seq_items == n
```
Comment (jreback) [inline pandas/tests/indexes/multi/test_formats.py:121]: prob better to make a new test as easier to see / run

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexes/multi/test_formats.py (hunk 3 above)
Comment (skvrahul) [reply pandas/tests/indexes/multi/test_formats.py]: Got it. Moved this to it's own test case

**Comment by skvrahul** [comment]
@jreback Added tests and moved one of the tests to its own TC.

File: doc/source/whatsnew/v1.3.0.rst [hunk 4]
```diff
@@ -144,8 +144,6 @@ Performance improvements
 Bug fixes
 ~~~~~~~~~
 
--
--
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.3.0.rst:147]: can you undo this

Comment (skvrahul) [reply doc/source/whatsnew/v1.3.0.rst]: Done. Have reverted the changes.

File: pandas/io/formats/printing.py (hunk 2 above)
Comment (jreback) [inline pandas/io/formats/printing.py:385]: do we have a case of max_seq_items==0 ? (I think this is a valid value), < 0 is restricted i am pretty sure?

**Review by jreback — CHANGES_REQUESTED** [review]
pls also merge master

File: pandas/io/formats/printing.py (hunk 2 above)
Comment (skvrahul) [reply pandas/io/formats/printing.py:385]: Yes we do. Max_seq_items ==0 implies unlimited if I'm not mistaken

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @skvrahul very nice

**Pull request merged**